opensearch_py_ml/ml_commons/ml_commons_client.py

```diff
@@ -568,6 +568,29 @@ def unload_model(self, model_id: str, node_ids: List[str] = []) -> object:
             body=API_BODY,
         )
 
+    def predict(self, algorithm_name: str, model_id: str, predict_object: dict) -> dict:
```

Comment:
We have a method [generate_model_inference](https://github.com/opensearch-project/opensearch-py-ml/blob/39284d6e99cd9b449bcfe99511c68a63d897ca78/opensearch_py_ml/ml_commons/ml_commons_client.py#L501-L517), which is in fact the predict functionality. We can just invoke this method here? 

We could change the method name `generate_model_inference` to predict, but for backward compatibility keep the method as it is.

Reply:
`generate_model_inference ` and `predict` methods are not the same because:
1) URLs are not the same
2) `generate_model_inference ` doesn't use `algorithm_name`

Reply:
could you please take a look at [the issue description](https://github.com/opensearch-project/opensearch-py-ml/issues/287)?

Reply:
I see @dhrubo-os, thank you for the feedback!

Reply:
So aren't we going to remove the `algorithm_name` here?

Reply:
Good point. I think it's better to keep this as a optional parameter for the BWC as before we needed to have algorithm_name for model prediction.


@Yerzhaisang currently algorithm_name isn't optional here. Can we do something like `def predict(self, model_id: str, predict_object: dict, algorithm_name: str = None) -> dict:` ?

Reply:
done